src/Featurizers/Components/NormUpdaters.h

```diff
@@ -0,0 +1,413 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include <limits>
+#include "../../Traits.h"
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+namespace Components {
+
+namespace TypeSelector {
+// max norm type selector is created for selecting different types for max norm based on input type
+// when input type is integer, uint64 is chosen for performance
+// when input type is numeric, long double is chosen
+template <typename T>
+struct MaxNormTypeSelector {
+    using type = std::uint64_t;
+};
+template <>
+struct MaxNormTypeSelector<std::float_t> {
+    using type = long double;
+};
+
+template <>
+struct MaxNormTypeSelector<std::double_t> {
+    using type = long double;
+};
+
+}
+
+#if (defined __clang__)
+#   pragma clang diagnostic push
+#   pragma clang diagnostic ignored "-Wfloat-equal"
+#   pragma clang diagnostic ignored "-Wdouble-promotion"
+#endif
+
+namespace Details{

```

Comment:
These shouldn't be in the Details namespace, but probably should go into a "NormUpdaters" namespace. (just change the name)